Jobs/GenerateTag.yml

```diff
@@ -0,0 +1,69 @@
+## @file
+# Template file used to generate tags on ADO. This template requires that the
+# consumer specifies this repository as a resource named mu_devops.
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+parameters:
+  - name: base_version
+    displayName: The base version.
+    type: string
+    default: ""
+  - name: git_name
+    displayName: Name to use for creating tag.
+    type: string
+    default: ""
+  - name: git_email
+    displayName: Email to use for creating tag.
+    type: string
+    default: ""
+  - name: notes_file
+    displayName: Path to the notes file to generate.
+    type: string
+    default: "ReleaseNotes.md"
+
+jobs:
+  - job: Create_Release_Tag
+    steps:
+      - checkout: self
+        clean: true
+        fetchTags: true
+        persistCredentials: true
+        path: "target"
+
+      - checkout: mu_devops
+        path: "mu_devops"
+        fetchDepth: 0
+
+      - template: ../Steps/SetupPythonPreReqs.yml
+        parameters:
+          install_pip_modules: false
+
+      - script: |
+          python -m pip install --upgrade pip
+          pip install GitPython
+        displayName: "Install Dependencies"
+
+      - script: |
+          git config --global user.name "${{ parameters.git_name }}"
+          git config --global user.email "${{ parameters.git_email }}"
+        displayName: "Setup Git"
+
+      - script: |
+          python mu_devops/Scripts/TagGenerator.py -r target/ -b ${{ parameters.base_version }} -v --tagvar tag_name --notes target/${{ parameters.notes_file }} --url $(Build.Repository.Uri)

```

Comment:
Should this be `$(tag_name)`?

Reply:
No, perhaps a bit hacky but I am actually passing the name of the variable for the python script to set it. A bit odd perhaps but I think its cleaner then trying to parse it out from the script output personally.

Reply:
ah, name of the variable :) that is good to know.